lib/jsonapi/view.ex

```diff
@@ -104,10 +124,37 @@ defmodule JSONAPI.View do
         def type, do: raise("Need to implement type/0")
       end
 
-      def attributes(data, conn) do
-        hidden = hidden(data)
+      defp requested_fields_for_type(%Conn{assigns: %{jsonapi_query: %{fields: fields}}} = conn) do
+        fields[type()]
+      end
+
+      defp requested_fields_for_type(_conn), do: nil
+
+      defp fields_for_type(requested_fields, fields) when requested_fields in [nil, %{}],
+        do: fields
+
+      defp fields_for_type(requested_fields, fields) do
+        fields
+        |> MapSet.new()
+        |> MapSet.intersection(MapSet.new(requested_fields))
+        |> MapSet.to_list()
+      end
+
+      @spec visible_fields(map(), conn :: nil | Conn.t()) :: map()
+      def visible_fields(data, conn) do
+        all_fields =
+          conn
+          |> requested_fields_for_type()
+          |> fields_for_type(fields())
+
+        hidden_fields = hidden(data)
 
-        visible_fields = fields() -- hidden
+        all_fields -- hidden_fields
+      end
+
+      @spec attributes(map(), conn :: nil | Conn.t()) :: map()
+      def attributes(data, conn) do
+        visible_fields = visible_fields(data, conn)
```

Comment:
What do you think about structuring this line like so ... `visible_fields = all_fields(conn) - hidden_fields(data)`?  To me it seems `visible_fields` is private and is already tested through your `PostView.attributes` test.

Reply:
I think `visible_fields` ought to be public. Since `attributes/2` is overridable we ought to provide the building blocks for sparse fieldsets to our users.

Reply:
That could be a good idea 👍 Do you think we should improve documentation / make a note of this behaviour somewhere?

Reply:
Sure thing!